File: src/api/MaterialAdvisor/MaterialAdvisor.Application/Models/KnowledgeChecks/KnowledgeCheckListItem.cs
namespace MaterialAdvisor.Application.Models.KnowledgeChecks;

public class KnowledgeCheckListItem
{
    public Guid Id { get; set; }

    public string Name { get; set; } = null!;

    public DateTime StartDate { get; set; }

    public DateTime? EndDate { get; set; }

    public ushort? Time { get; set; }

    public byte? MaxAttempts { get; set; }

    public double? Score { get; set; }

    public double PassScore { get; set; }

    public double MaxScore { get; set; }

    public byte? UsedAttempts { get; set; }

    public bool IsSubmitted { get; set; }

    public bool IsVerified { get; set; }

    public bool IsAttemptOverrided { get; set; }

    public bool IsManualOnlyVerification { get; set; }

    public bool HasAnswersToVerify { get; set; }
}

File: src/app/material-advisor/src/app/models/knowledge-check/KnowledgeCheckListItem.ts
import { GUID } from "@shared/types/GUID";

export interface KnowledgeCheckListItem {
  id: GUID;
  name: string;
  time: number;
  startDate: Date;
  endDate?: Date;
  usedAttempts: number;
  maxAttempts?: number;
  isSubmitted: boolean;
  isVerified: boolean;
  passScore: number;
  maxScore: number;
  score?: number;
}